integration_test/list_test.go

```diff
@@ -42,3 +47,260 @@ func TestKrewList(t *testing.T) {
 	}
 	// TODO(ahmetb): install multiple plugins and see if the output is sorted
 }
+
+func TestKrewListJSON(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest", filepath.Join("testdata", "foo.yaml"),
+		"--archive", filepath.Join("testdata", "foo.tar.gz")).
+		RunOrFail()
+
+	expected := []byte("[\n    {\n        \"Name\": \"foo\",\n        \"Version\": \"v0.1.0\"\n    }\n]\n")
+
+	eventualList := test.WithIndex().Krew("list", "-o", "json").RunOrFailOutput()
+	if diff := cmp.Diff(eventualList, expected); diff != "" {
+		t.Fatalf("'list' output doesn't match:\n%s", diff)
+	}
+
+	var Plugin []struct {
+		Name    string
+		Version string
+	}
+
+	err := json.Unmarshal(eventualList, &Plugin)
+	if err != nil || Plugin[0].Name != "foo" || Plugin[0].Version != "v0.1.0" {
```

Comment:
just do a slice cmp by having an `expected` slice?

Reply:
We wanted confirmation that valid JSON was being output (it only seems sane at that point to also test the accuracy)

Reply:
Also in YAML's(?) case the error result was always nil, regardless of failure.